architecture/2024-05-radius-on-dapr.md

```diff
@@ -0,0 +1,324 @@
+# Radius on Dapr
+
+* **Status**: Pending/Approved
+* **Author**: Ryan Nowak (`@rynowak`)
+
+## Overview
+
+The design doc proposes replacing some existing plumbing in Radius with [Dapr](https://dapr.io) and taking an install-time dependency on Dapr. Radius needs a database and a message queue to do its work. Dapr provides implementations of its [building block APIs](https://docs.dapr.io/concepts/building-blocks-concept/) that work across self-hostable open-source technology and managed cloud services. Compared to developing our own data layer, adopting Dapr will accellerate development and production-readiness of Radius because Dapr is already mature and production-hardened. Dapr provides SDKs for most languages, including Go and C# so it can be used in the Deployment Engine.
+
+Dapr also brings with it a powerful [workflow](https://docs.dapr.io/developing-applications/building-blocks/workflow/) programming model that can be used to write reliable and scalable distributed systems workflows. This is ideal for building more complex background processing in Radius, and also ideal for us to offer extensibility to users.
+
+Dapr is well-supported both inside and outside of Kubernetes. Adopting a technology like Dapr will help Radius work well regardless of the hosting scenario.
```

Comment:
If we decide to do **serverless**, would **Dapr** also be able to help there too?

Reply:
Yes absolutely.